_posts/2017-12-20-Virtual-Reality.md

```diff
@@ -0,0 +1,270 @@
+Hardware that supports MR take the form of glasses and headsets with cameras and other sensors.
+These sensors are used to map the physical 3D space so the virtual objects know where they are situated and know what is in the real world.
+This is where you get the holographic experiences, even tough they aren't real holograms, they look and feel exactly as you would expect them to.
+
+<img alt="Mixed reality" src="{{ '/img/virtualreality/banner.jpg' | prepend: site.baseurl }}" class="image fit" style="margin:0px auto; max-width: 600px;">
+
+
+## Hardware and current pain points
+Current devices are a big step up from what we had even a few years ago, but there are still many improvements to be made.
+The high end VR devices require beefy computers equipped with high-end graphics cards.
+The headsets requiring a smartphone offer a more mobile experience but are still limited in the visual effects department.
+For a virtual reality experience to feel smooth and natural it needs to be rendered at a minimum of 90 frames per second.
+If the device can not keep up the user might experience VR sickness, similar to motion sickness.
+Because our eyes are so close to the screen and the resolution our eyes can perceive is so high, pixels become visible, detracting from the experience.
+When looking closely, the screen looks like a raster.
+This can be solved with a higher resolution but that again requires more computational power.
+This much content at such a high pace also requires a lot of bandwidth and low latency.
+
+Another deterrent for VR headsets is the extensive setup required to make them work.
+High end devices require base stations or tracking cameras to accurately measure movement.
+There are devices with sensors built in so they are not dependent on external base stations, this technique is called inside-out tracking.
+
+## Creating realities
+Making software that is AR/VR/MR enabled can be quite different from traditional business application development.
+These type of applications are often heavily dependent on visual elements and can require extra artistic knowledge.
+For virtual and mixed reality game engines can be used to create "games" to create these experiences.
+
+These are some of the most popular and game engines that are free for personal user:
+- Unity
+- Unreal Engine
+- CryEngine
+
+Think of these as sort of photoshop for software.
+We can leverage the editors to see what we're working on and make changes immediately.
+These tools are ideal for Virtual Reality.
+Both Unity and Unreal Engine have plugins available for Augmented Reality.
+
+The often more business targeted applications are often augmented reality enabled applications.
+Both Apple and Google have made big efforts to support this on their respective OSes and basically every smartphone with a camera is now AR enabled.
+Apple's ARKit and Google's ARCore are the respective Augmented reality APIs for their IOS and Android platforms.
```

Comment:
Augmented Reality or augmented reality (currently it's one with a capital and one without a capital)
iOS